pytorch_lightning/accelerators/ddp_backend.py

```diff
@@ -149,15 +261,7 @@ def set_world_ranks(self, process_idx):
         self.trainer.world_size = self.trainer.num_nodes * self.trainer.num_processes
 
     def model_to_device(self, model, process_idx, is_master):
-        gpu_idx = process_idx
-
-        # when using ddp, the master process (proc 0) continues running as the main one
-        # this means that the local rank will always be 0
-        # (even if cuda visible devices has other visible gpus)
-        # this means that the master process needs to pull the 0th visible index as the device number
-        if is_master:
-            available_gpus = os.environ['CUDA_VISIBLE_DEVICES'].split(',')
-            gpu_idx = int(available_gpus[self.trainer.local_rank])
+        gpu_idx = int(os.environ.get('PL_DDP_PID', process_idx))
```

Comment:
`is_master` is never used here. it seems like ddp_train doesn't need the self.mode == 'ddp' check anymore

Reply:
Good catch! 
might be a weird merge artifact... let me check.